test/robustness/within/lambda_partition.hpp

```diff
@@ -0,0 +1,98 @@
+    F m_f;
+
+    explicit adapt_partition_visitor(F&& f)
+        : m_f(std::move(f))
+    {}
+
+    template <typename ...Ts>
+    decltype(auto) apply(Ts&& ...is) const
+    {
+        return m_f(std::forward<Ts>(is)...);
+    }
+};
+
+template
+<
+    typename BoxType,
+    typename ForwardRange1,
+    typename ForwardRange2
+>
+bool lambda_partition(ForwardRange1 const& forward_range1,
+                        ForwardRange2 const& forward_range2,
+                        const std::function
+                            <
+                                void(BoxType&,
+                                     typename boost::range_value<ForwardRange1>::type const&)
+                            >& expand_policy1,
+                        const std::function
+                            <
+                                bool(BoxType const&,
+                                     typename boost::range_value<ForwardRange1>::type const&)
+                            >& overlaps_policy1,
+                        const std::function
+                            <
+                                void(BoxType&,
+                                     typename boost::range_value<ForwardRange2>::type const&)
+                            >& expand_policy2,
+                        const std::function
+                            <
+                                bool(BoxType const&,
+                                     typename boost::range_value<ForwardRange2>::type const&)
+                            >& overlaps_policy2,
+                        const std::function
+                            <
+                                bool(typename boost::range_value<ForwardRange1>::type const&,
+                                     typename boost::range_value<ForwardRange2>::type const&)
+                            >& visitor,
+                        const std::function
+                            <
+                                void(BoxType const&, int level)
+                            >& box_visitor = [](auto const&, int) {})
+{
+    adapt_partition_visitor<decltype(visitor)> av(visitor);
+    adapt_partition_visitor<decltype(expand_policy1)> aev1(expand_policy1);
+    adapt_partition_visitor<decltype(expand_policy2)> aev2(expand_policy2);
+    adapt_partition_visitor<decltype(overlaps_policy1)> aov1(overlaps_policy1);
+    adapt_partition_visitor<decltype(overlaps_policy2)> aov2(overlaps_policy2);
+    adapt_partition_visitor<decltype(box_visitor)> apbv(box_visitor);
+
+    // TODO: the "16" could be part of an optional options structure
+    // The all_policies (it is never run in a different way) could be
+    // a compiletime part of that options structure as well.
+    return partition
+        <
+            BoxType,
+            detail::partition::include_all_policy,
+            detail::partition::include_all_policy
+        >::apply(forward_range1, forward_range2, av,
+                 aev1, aov1,
+                 aev2, aov2, 16, apbv);
```

Comment:
maybe add a comment that 16 is the end of the iteration (the max number of points in a partition box)

Reply:
I changed it to an options structure, as I planned earlier.
Is something like this fine?
:heavy_check_mark: